### PR TITLE
[Agregar validación de precondiciones...] Guava preconditions implementation

### PR DESCRIPTION
**Pull request by jhonmunoz**
## Summary

Adds Guava for precondition validations on models

## Trello Card
https://trello.com/c/Ko1ZhSes/29-agregar-validaci%C3%B3n-de-precondiciones-a-los-setters

### CODE REVIEW

**Review by rabinovichm — CHANGES_REQUESTED** [review]
Add validations according to the [books](https://trello.com/c/BbeRyYdW/14-crear-modelo-de-book) and [users](https://trello.com/c/J9kZfwst/19-crear-modelo-de-user) model. For example, a book cannot have less than 1 page.

File: src/main/java/com/wolox/training/models/User.java [hunk 1]
```diff
@@ -52,24 +54,22 @@ public String getUsername() {
...
     }
 
     public LocalDate getBirthDate() {
         return birthDate;
     }
 
-    public void setBirthDate(final LocalDate birthDate) {
-        this.birthDate = birthDate;
-    }
+    public void setBirthDate(final LocalDate birthDate) { this.birthDate = checkNotNull(birthDate); }
```
Comment (rodrigofranc) [inline src/main/java/com/wolox/training/models/User.java]: It should be before now

**Review by rodrigofranc — CHANGES_REQUESTED** [review]
All strings should be validate for not be blank

**Review by rodrigofranc — CHANGES_REQUESTED** [review]
Add validation for empty string in setters

**Review by rodrigofranc — APPROVED** [review]

**Review by rabinovichm — APPROVED** [review]

**Pull request merged**